lib/storage.ts

```diff
@@ -0,0 +1,68 @@
+import { StorageInfo } from "@/types/storage";
+import * as fs from 'fs';
+
+function parsePasswdFile(fileContent: string): { [id: string]: string } {
+    const lines = fileContent.split('\n');
+    const idToUsername: { [id: string]: string } = {};
+
+    for (const line of lines) {
+        const fields = line.split(':');
+        const username = fields[0];
+        const id = fields[2];
+
+        if (username && id) {
+            idToUsername[id] = username;
+        }
+    }
+
+    return idToUsername;
+}
+
+async function fetchUsernames(): Promise<{ [id: string]: string }> {
+    try {
+        const fileContent = await fs.promises.readFile('/etc/passwd', 'utf-8');
+        const idToUsername = parsePasswdFile(fileContent);
+        return idToUsername;
+    } catch (error) {
+        console.error('There was a problem fetching the usernames:', error);
+        throw error;
+    }
+}
+
+// Call fetchUsernames() and use the returned dictionary in your code
+
+// Use idToUsername in your code
+
+async function fetchFileContent(url: string = 'http://10.176.52.3:5000/show'): Promise<string> {
```

Comment:
Seems better to use an environmental variable to load the backend URL rather than hard-coding it.

Reply:
Thank you for the suggestion. I've added environment variables for url and formatted with prettier.